app/models/embed/purl/resource_file.rb

```diff
@@ -3,27 +3,20 @@
 module Embed
   class Purl
     class ResourceFile
-      # @param [String] druid the identifier of the resource this file belongs to.
-      # @param [String] description the label for this file
-      # @param [Nokogiri::XML::Element] file
-      # @param [Dor::RightsAuth] rights
-      def initialize(druid, description, file, rights)
-        @druid = druid
-        @description = description
-        @file = file
-        @rights = rights
-        @index = nil
+      def initialize(attributes = {})
+        self.attributes = attributes
+        yield(self) if block_given?
       end
 
-      attr_accessor :index
-
-      def label
-        @description
+      def attributes=(hash)
+        hash.each do |key, value|
+          send("#{key}=", value)
```

Comment:
Maybe a bit paranoid, but should we consider changing this to `#public_send`?

Reply:
Sure.